photon-api/src/main/scala/com/linkedin/photon/ml/algorithm/RandomEffectCoordinate.scala

```diff
@@ -228,12 +227,16 @@ object RandomEffectCoordinate {
       randomEffectOptimizationProblem: RandomEffectOptimizationProblem[Function],
       initialRandomEffectModelOpt: Option[RandomEffectModel]): (RandomEffectModel, RandomEffectOptimizationTracker) = {
 
-    // All 3 RDDs involved in these joins use the same partitioner
+    // All 3 RDDs involved in the joins below use the same partitioner
+
+    // Optimization problems are created for each entity with a projector, and thus guaranteed to match active data
+    // exactly (see RandomEffectDataset.apply)
     val dataAndOptimizationProblems = randomEffectDataset
       .activeData
       .join(randomEffectOptimizationProblem.optimizationProblems)
 
-    // Left join the models to data and optimization problems for cases where we have a prior model but no new data
+    // Left join the models to the (data, optimization problem) tuple for cases where we have a prior model but no new
```

Comment:
To confirm, does a prior model only come from warm start? If so, prior models and new data might not match and we can do left join. Do we still keep this model and output it in this case?

Reply:
That's right - if Photon ML trains a warm-start model, it will keep existing models for which no new data is present.